src/TodoList.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Dispatch, SetStateAction, useEffect, useState,
+} from 'react';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo[],
+  status: Status,
+  setTodos: Dispatch<SetStateAction<Todo[]>>,
+  setErrorMessage: Dispatch<SetStateAction<string>>,
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos, status, setTodos, setErrorMessage,
+}) => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
```

Comment:
we don't need the separate state for visible todos, let's just filter them and return in the variable filtered array of todos